modules/planning/common/obstacle_blocking_analyzer.cc

```diff
@@ -0,0 +1,178 @@
+  if (!obstacle->IsStatic() ||
+      obstacle->speed() > block_obstacle_min_speed) {
+    ADEBUG << " - It is non-static.";
+    return false;
+  }
+
+  // Obstacle is behind ADC.
+  if (obstacle->PerceptionSLBoundary().start_s() <=
+      adc_sl_boundary.end_s()) {
+    ADEBUG << " - It is behind ADC.";
+    return false;
+  }
+
+  // Obstacle is far away.
+  constexpr double kAdcDistanceThreshold = 15.0;  // unit: m
+  if (obstacle->PerceptionSLBoundary().start_s() >
+      adc_sl_boundary.end_s() + kAdcDistanceThreshold) {
+    ADEBUG << " - It is too far ahead.";
+    return false;
+  }
+
+  // Obstacle is too close.
+  if (adc_sl_boundary.end_s() + min_front_sidepass_distance >
+      obstacle->PerceptionSLBoundary().start_s()) {
+    ADEBUG << " - It is too close to side-pass.";
+    return false;
+  }
+
+  // Obstacle is not blocking our path.
+  if (!ObstacleIsBlockingDrivingPath(reference_line, obstacle)) {
+    ADEBUG << " - It is not blocking our way.";
+    return false;
+  }
+
+  // Obstacle is blocked by others too.
+  if (enable_obstacle_blocked_check && !IsParked(reference_line, obstacle)) {
+    for (const auto* other_obstacle : path_decision.obstacles().Items()) {
+      if (other_obstacle->Id() == obstacle->Id()) {
+        continue;
+      }
+      if (other_obstacle->IsVirtual()) {
+        continue;
+      }
+      if (other_obstacle->PerceptionSLBoundary().start_l() >
+              obstacle->PerceptionSLBoundary().end_l() ||
+          other_obstacle->PerceptionSLBoundary().end_l() <
+              obstacle->PerceptionSLBoundary().start_l()) {
+        // not blocking the backside vehicle
+        continue;
+      }
+      double delta_s = other_obstacle->PerceptionSLBoundary().start_s() -
+                       obstacle->PerceptionSLBoundary().end_s();
+      if (delta_s < 0.0 || delta_s > kAdcDistanceThreshold) {
+        continue;
+      }
+
+      // TODO(All): Fix the segmentation bug for large vehicles, otherwise
+      // the follow line will be problematic.
+      ADEBUG << " - It is blocked by others, too.";
+      return false;
+    }
+  }
+
+  ADEBUG << "IT IS BLOCKING!";
+  return true;
+}
+
+
+bool ObstacleIsBlockingDrivingPath(
+    const ReferenceLine& reference_line, const Obstacle* obstacle) {
+  const double driving_width =
+      reference_line.GetDrivingWidth(obstacle->PerceptionSLBoundary());
+  const double adc_width =
+      VehicleConfigHelper::GetConfig().vehicle_param().width();
+  ADEBUG << " (driving width = " << driving_width
+         << ", adc_width = " << adc_width << ")";
+  if (driving_width >
+      adc_width + FLAGS_static_decision_nudge_l_buffer + 0.4) {
+    // TODO(jiacheng): make this a GFLAG:
+    // side_pass_context_.scenario_config_.min_l_nudge_buffer()
+    ADEBUG << "It is NOT blocking our path.";
+    return false;
+  } else {
+    ADEBUG << "It is blocking our path.";
+    return true;
+  }
+  return false;
+}
+
+
+bool IsParked(const ReferenceLine& reference_line,
```

Comment:
Change this function name as it will be used without a class. Suggest "IsParkedVechicle", etc.